### PR TITLE
add code to C semantics for annotation generation

### PR DESCRIPTION
**Pull request by dwightguth**
@andreistefanescu @chathhorn please review

### CODE REVIEW

File: semantics/language/translation/init.k [hunk 1]
```diff
@@ -260,5 +261,26 @@ module C-TRANSLATION-INIT
      syntax KItem ::= "reportSuccess"
      rule <k> reportSuccess => .K ...</k>
           <result-value> _ => 0 </result-value>
+          <generated-annotations> .List </generated-annotations>
+     rule <k> reportSuccess => writeSpecModule(#open(S +String ".spec.k")) ...</k>
```
Comment (andreistefanescu) [inline semantics/language/translation/init.k]: maybe ```S +String "_spec.k"```?

**Comment by andreistefanescu** [comment]
lgtm

**Comment by andreistefanescu** [comment]
lgtm

File: semantics/language/translation/init.k (hunk 1 above)
Comment (dwightguth) [inline semantics/language/translation/init.k]: we could do that, but that will lead to files called foo.c_spec.k or foo.h_spec.k, which I am inclined to think looks worse than foo.c.spec.k or foo.h.spec.k.

**Comment by dwightguth** [comment]
Jenkins: test this please

**Pull request merged**